src/quantization.cpp

```diff
@@ -0,0 +1,89 @@
+#include <migraphx/program.hpp>
+#include <migraphx/quantization.hpp>
+#include <migraphx/stringutils.hpp>
+#include <migraphx/instruction.hpp>
+#include <migraphx/op/fp_conversion.hpp>
+#include <migraphx/target.hpp>
+#include <migraphx/env.hpp>
+#include <migraphx/ranges.hpp>
+#include <migraphx/time.hpp>
+#include <migraphx/iterator_for.hpp>
+#include <iostream>
+#include <sstream>
+#include <algorithm>
+#include <utility>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+
+instruction_ref convert_fp32_fp16(program& prog, instruction_ref& ins)
+{
+    assert(ins->get_shape().type() == shape::float_type);
+    assert(ins->name().front() == '@');
+    instruction_ref ins_fp16{};
+    if(ins->name() == "@literal")
+    {
+        std::vector<float> values;
+        auto l_fp32 = ins->get_literal();
+        shape s     = ins->get_shape();
+        l_fp32.visit([&](auto val) { values.assign(val.begin(), val.end()); });
```

Comment:
also, we need to cast float into __f16 (gcc) here, so instead of "std::vector<float> value", should it be: "std::vector<__f16> value" ?  I don't know what is the equivalent half float type in clang.

Reply:
Actually, gcc probably only supports half precision on ARM machines.

Reply:
There is no need for a cast or explicit constructor because the `assign` method will explicitly call the constructor for `half`. Otherwise there would be a compile error here.

Reply:
If the assumption is that it is always `float` then you could use `l_fp32.get<float>()` instead of using the visitor.